sqlest/src/main/scala/sqlest/executor/Database.scala

```diff
@@ -224,6 +229,38 @@ case class Transaction(database: Database) extends Session(database) {
       }
     }
 
+  def executeCommandReturningKeys[T](command: Command)(implicit columnType: ColumnType[T]): RowCountAndKeys[T] =
+    withConnection { connection =>
+      val (preprocessedCommand, sql, argumentLists) = database.statementBuilder(command)
+      val startTime = new DateTime
+      try {
+        val preparedStatement = prepareStatement(connection, preprocessedCommand, sql, argumentLists)
+        try {
+          val keys = List[T]()
+          val result = preparedStatement.executeUpdate
+          val rs = preparedStatement.getGeneratedKeys
+          val extractor = extract[RowCountAndKeys[T]](
+            rowsUpdated = extractConstant[Int](result),
+            keys = IndexedColumn[T](1).asList
+          )
+          val countAndKeys = extractor.extractHeadOption(ResultSetIterable(rs))
+          val endTime = new DateTime
+          logger.info(s"Ran sql in ${endTime.getMillis - startTime.getMillis}ms: ${logDetails(connection, sql, argumentLists)}")
+          countAndKeys.getOrElse {
+            throw new SQLException("No generated keys found")
+          }
+        } finally {
+          try {
+            if (preparedStatement != null) preparedStatement.close
+          } catch { case e: SQLException => }
```

Comment:
hey hey :) true, but I'm surprised it's not logged at debug or something? You could be right though